language/stdlib/modules/libra_transaction_timeout.mvir

```diff
@@ -0,0 +1,49 @@
+module LibraTransactionTimeout {
+    import 0x0.LibraTimestamp;
+
+    resource TTL {
+        // Only transactions with timestamp in between block time and block time + duration would be accepted.
+        duration: LibraTimestamp.Duration,
+    }
+
+    public initialize() {
+        let duration: LibraTimestamp.Duration;
+        let timeout: Self.TTL;
+
+        // Only callable by the Association address
+        assert(get_txn_sender() == 0xA550C18, 1);
+
+        // Currently set to 1day.
+        duration = LibraTimestamp.new_duration(86400);
+        timeout = TTL {duration: move(duration)};
+        move_to_sender<TTL>(move(timeout));
+        return;
+    }
+
+    public set_timeout(new_duration: LibraTimestamp.Duration) acquires TTL {
+        let timeout: &mut Self.TTL;
+
+        // Only callable by the Association address
+        assert(get_txn_sender() == 0xA550C18, 1);
+
+        timeout = borrow_global_mut<TTL>(0xA550C18);
+        *(&mut move(timeout).duration) = move(new_duration);
+        return;
+    }
+
+    public is_valid_transaction_timestamp(timestamp: u64): bool acquires TTL {
```

Comment:
I can't shake the feeling that the abstractions in the `LibraTimstamp` API make it really hard to understand/write this code... I'm wondering if we shouldn't just start with raw integers and add these layers of abstraction later.